### DISCUSSION TITLE
AppNotificationManager throw exception.

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/WindowsAppSDK#3472] [Q&A]
It`s a very simple code

![image](https://user-images.githubusercontent.com/15758574/222398752-39455665-2d97-4e76-ac54-5d619615b988.png)

 not found element.. what`s that mean? 

papckage,appxmainfest
![image](https://user-images.githubusercontent.com/15758574/222399347-898215ef-7cec-4fa1-8e89-631bcf879391.png)

**Comment by castorix** [comment]
Did you follow all the steps from  [Quickstart: App notifications in the Windows App SDK](https://github.com/MicrosoftDocs/windows-dev-docs/blob/docs/hub/apps/windows-app-sdk/notifications/app-notifications/app-notifications-quickstart.md) ?

**Reply by pmpurifoy** [reply]
Multiple issues to address here:
1. How to solve your ERROR_NOT_FOUND issue.
1a.) Call AppNotificationManger.Register to register your app for AppNotifications. 
1b.) Your "Arguments" parameter for your ExeServer is for PushNotifications.
From here: https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/notifications/push-notifications/push-quickstart#step-2-add-your-com-activator-to-your-apps-manifest

You need to replace it with the arguments for AppNotifications: Arguments="----AppNotificationActivated:"

2. Need to register the process from the package folder. 

You need to change the path for your exe. When you package your exe, it should be under the package folder, so replace demo.exe with "demo/demo.exe".

3. Make sure you register an activation handler to receive user interaction (You don't have to do this yet since you are just testing show, but this will be important for you soon).

**Reply by ghost** [reply]
> Did you follow all the steps from [Quickstart: App notifications in the Windows App SDK](https://github.com/MicrosoftDocs/windows-dev-docs/blob/docs/hub/apps/windows-app-sdk/notifications/app-notifications/app-notifications-quickstart.md) ?

Thinks.
I`m doing this follow this doc and toast it`s working.

but, I have another question.
this demo code , does not has the 'Extension' element in Package.appxmainfest.  
but local toast working yet.
![image](https://user-images.githubusercontent.com/15758574/222647659-ccf7d79a-a33d-48c4-951e-c5dbba178bc5.png)
I think the 'Extension' element it`s about  Azure`s NotificationHubs?
Am I right?

**Reply by castorix** [reply]
Extensions are described in MSDN : https://learn.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/extensions
(but I never tested)
